### PR TITLE
Combine namedFiles and archive of metadata files into webTestFiles.

### PR DESCRIPTION
**Pull request by DrMarcII**

### CODE REVIEW

File: go/metadata/web_test_files.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+)
+
+// WebTestFiles defines a set of namedFiles located either in the runfiles directory or
+// in an archive file located in the runfiles directory of the test.
+type WebTestFiles struct {
+	ArchiveFile string            `json:"archiveFile,omitempty"`
+	NamedFiles  map[string]string `json:"namedFiles"`
+
+	mu            sync.Mutex
+	extractedPath string
```
Comment (juangj) [inline go/metadata/web_test_files.go]: Add a comment explaining the purpose of this?

File: go/metadata/web_test_files.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+}
+
+// Only works correctly on normalized WeebTestFiles slices
```
Comment (juangj) [inline go/metadata/web_test_files.go]: s/Weeb/Web/

File: go/metadata/web_test_files.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/bazelbuild/rules_webtesting/go/util/bazel"
+)
+
+// WebTestFiles defines a set of namedFiles located either in the runfiles directory or
+// in an archive file located in the runfiles directory of the test.
```
Comment (juangj) [inline go/metadata/web_test_files.go]: So, if ArchiveFile == "" then the NamedFiles are located in the runfiles?

File: web/internal/web_test_named_file.bzl [hunk 4]
```diff
@@ -31,7 +31,7 @@ def _web_test_named_file_impl(ctx):
     patch = ctx.outputs.web_test_metadata
 
   content = """{
-  "namedFiles": {"%s": "%s"}
+  "webTestFiles": ["namedFiles": {"%s": "%s"} }]
```
Comment (juangj) [inline web/internal/web_test_named_file.bzl]: Missing a '{' here.

File: go/metadata/web_test_files.go (hunk 1 above)
Comment (DrMarcII) [inline go/metadata/web_test_files.go]: Done.

File: go/metadata/web_test_files.go (hunk 2 above)
Comment (DrMarcII) [inline go/metadata/web_test_files.go]: Done.

File: web/internal/web_test_named_file.bzl (hunk 4 above)
Comment (DrMarcII) [inline web/internal/web_test_named_file.bzl]: Done.

**Pull request merged**